Snippets/Snippets_5/RavenDB/ConfiguringTransactionRecoveryStorage.cs

```diff
@@ -7,20 +10,55 @@
 
     class ConfiguringTransactionRecoveryStorage
     {
-        public void Foo()
+        public void ConfiguringTransactionRecoveryStorageBasePath()
         {
-            #region ConfiguringTransactionRecoveryStorage
+            #region ConfiguringTransactionRecoveryStorageBasePath
 
-            string transactionRecoveryPath = "path to transaction recovery storage unique per endpoint";
-            DocumentStore myDocumentStore = new DocumentStore
-            {
-                TransactionRecoveryStorage = new LocalDirectoryTransactionRecoveryStorage(transactionRecoveryPath)
-            };
-            // configure document store properties here and initialize
+            // Same value for all endpoints; must be writeable
+            string transactionRecoveryBasePath = "%LOCALAPPDATA%"; 
+            
+            BusConfiguration busConfiguration = new BusConfiguration();
+            busConfiguration.UsePersistence<RavenDBPersistence>()
+                .SetTransactionRecoveryStorageBasePath(transactionRecoveryBasePath);
 
+            #endregion
+        }
+
+        public void CustomizingDocumentStoreBeforeInit()
+        {
+            #region CustomizingRavenDocumentStoreBeforeInit
             BusConfiguration busConfiguration = new BusConfiguration();
             busConfiguration.UsePersistence<RavenDBPersistence>()
-                .SetDefaultDocumentStore(myDocumentStore);
+                .CustomizeDocumentStore(docStore =>
+                {
+                    // docStore can be customized here before NServiceBus
+                    // calls docStore.Initialize();
+                });
+            #endregion
+        }
+
+        public void ManualDtcSettingExample()
+        {
+            string UrlToRavenDB = "http://localhost:8080";
+
+            #region RavenDBManualDtcSettingExample
+            // Value must uniquely identify endpoint on the machine and remain stable on process restarts
+            Guid resourceManagerId = Guid.Parse(ConfigurationManager.AppSettings["EndpointResourceManagerId"]);
```

Comment:
no need to show `ConfigurationManager.AppSettings` here. we are not trying to document how to read from config